proposal/proposal.tex

```diff
@@ -138,85 +138,71 @@ \subsection{Parsetree based}
 
-Furthermore, each PPX was a separate and isolated binary; with one exception:
-for type derivers, there was $ppx \_ deriving$, which would orchestrate the
-different derivers.
+Initially, each PPX was a separate binary. The one early exception was $ppx \_
+deriving$, which orchestrates multiple type-based derivers into a single
+rewriter.
 
-In 2018, $ppxlib$\cite{ppxlib} was invented to solve both problems and bring
-more benefits (pointed out \hyperref[benefits]{below}). It exposes one fixed
-AST, which all OCaml PPXs are defined against. It coordinates with the compiler
-releases to bump that AST version in parallel. That's how it has consolidated a
-PPX ecosystem, which stays up-to-date.
-
-The main problem in terms of the consolidated ecosystem is that bumping the AST
-breaks a few PPXs each time. To solve that situation, there was the idea to
-abstract the AST exposed by $ppxlib$ and keep that one stable. That idea was
-presented at the OCaml workshop 2019\cite{workshop2019}. However, the idea
-involved too much complexity and was abandoned. We now have other means to
-approach the stability problem, pointed out \hyperref[stability]{below}.
+In 2018, $ppxlib$\cite{ppxlib} was created, in part to solve these problems. It
+exposes a single version of the AST, which is kept up to date with the latest
+compiler and is backward-compatible with previous releases. This consolidates
+the PPX ecosystem and upgrades most PPXes ``for free''.
 
+The primary remaining problem is the few PPXes that get broken each time the AST
+changes. Our initial proposal, presented at the OCaml workshop
+2019\cite{workshop2019}, was to create a single AST that could contain fragments
+from multiple versions at once. This proved too complex and was subsequently
+abandoned. We now approach \hyperref[stability]{stability} with a combination of
+technical and social solutions.
 
 \subsection{Typedtree based}
 
-There have also been efforts to introduce typedtree based meta-programming:
-$MetaOCaml$\cite{metaocaml}. It's been a very valuable experience and might
-still become more some day.
-%A nice feature of typedtree-based meta-programming is that the typing of the
-%generated code is checked at the time of writing the meta-program. It comes at
-%the cost of having to deal with two unstable representations instead of one:
-%both the parsetree and the typedtree. (ToDo: I'll ask Olivier if this paragraph
-%is correct and sounds good to the MetaOCaml folks)
+There have also been efforts to introduce typedtree based meta-programming,
+especially $MetaOCaml$\cite{metaocaml}. MetaOCaml and PPXes are different
+notions of metaprogramming with different goals and expressivity. MetaOCaml is
+type-safe, staged metaprogramming that can be used to specialize values at
+compile-time or run-time. PPXes offer more general code generation than
+specializing values, but lack MetaOCaml's ability to evaluate the code and its
+integration with the type system.
```

Comment:
Thanks, that's way better! Should we also mention somewhere that MetaOCaml is only available on a fork of OCaml?
```suggestion
There have also been efforts to introduce typedtree based meta-programming,
especially $MetaOCaml$\cite{metaocaml}. MetaOCaml and PPXes are different
notions of metaprogramming with different goals and expressivity. MetaOCaml is
type-safe, staged metaprogramming that can be used to specialize values at
compile-time or run-time. PPXes offer more general code generation than
specializing values, but lack MetaOCaml's ability to evaluate the code and its
integration with the type system. MetaOCaml requires non-merged changes
to the compiler and therefore only works on a fork of OCaml.
```